var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -261,6 +261,8 @@ class PyTensorflow(Package, CudaPackage, ROCmPackage, PythonExtension):
     # https://github.com/protocolbuffers/protobuf/issues/10051
     # https://github.com/tensorflow/tensorflow/issues/56266
     depends_on("py-protobuf@:3.19", type=("build", "run"), when="@:2.11")
+    # Must be matching versions of py-protobuf and protobuf
+    depends_on("py-protobuf+cpp", type=("build", "run"), when="^py-protobuf@:4.21")
```

Comment:
Maybe use `conflicts("^py-protobuf~cpp")` so you don't have to repeat the condition, would it ever change in the py-protobuf package?